binderhub/builder.py

```diff
@@ -250,8 +250,26 @@ async def get(self, provider_prefix, _unescaped_spec):
         except Exception as e:
             await self.fail("Error resolving ref for %s: %s" % (key, e))
             return
+
         if ref is None:
-            await self.fail("Could not resolve ref for %s. Double check your URL." % key)
+            error_message = ["Could not resolve ref for %s. Double check your URL." % key]
+
+            split_key = key.split("/")
+            repo_provider = split_key[0].split(":")[0]
+            resolved_branch = split_key[-1]
+
+            if repo_provider == "gh":
```

Comment:
"gh" is already available as `provider_prefix` from the function. Could also use `isinstance(repo_provider, GitHubRepoProvider)` or `provider.name == "GitHub"`